File: ConduitCore/src/main/java/systems/conduit/core/commands/CommandManager.java
package systems.conduit.core.commands;

import systems.conduit.core.Conduit;

import java.util.Arrays;

public class CommandManager {

    public void loadDefaultCommands() {
        Conduit.getServer().ifPresent(server -> {
            server.getCommands().getDispatcher().register(new PluginsCommand().getCommand());
            server.getCommands().getDispatcher().register(new VersionCommand().getCommand());
        });
    }

    public void registerCommand(BaseCommand... commands) {
        Conduit.getServer().ifPresent(server -> Arrays.asList(commands).forEach(command -> server.getCommands().getDispatcher().register(command.getCommand())));
    }
}

File: src/main/kotlin/systems/conduit/core/commands/CommandManager.kt
package systems.conduit.core.commands

import systems.conduit.core.Conduit

class CommandManager {

    fun loadDefaultCommands() {
        //Conduit.server.getCommands().dispatcher.register(PluginsCommand().command)
        //Conduit.server.getCommands().dispatcher.register(VersionCommand().command)
    }

    fun registerCommand(vararg commands: BaseCommand?) {
        listOfNotNull(*commands).forEach { command -> Conduit.server.getCommands().dispatcher.register(command.command) }
    }
}
